src/providers/Context/Context.provider.jsx

```diff
@@ -13,29 +15,59 @@ function useVideosUpdate() {
   return useContext(VideoUpdateContext);
 }
 
+function useVideoDetailed() {
+  return useContext(VideoDetailedContext);
+}
+
+function reducer(state, action) {
+  switch (action.type) {
+    case 'grid':
+      return { videos: action.payload.data, video: null };
+    case 'detail':
+      return { videos: null, video: action.payload.data };
+    default:
+      return state;
+  }
+}
+
 const ContextProvider = ({ children }) => {
-  const [videos, setVideos] = useState(dataInit);
+  const [state, dispatch] = useReducer(reducer, { videos: dataInit });
 
   function searchVideos(e) {
     if (e.key === 'Enter') {
       e.preventDefault();
 
       if (e.target.value.length === 0) {
-        setVideos(dataInit);
+        dispatch({ type: 'grid', payload: { data: dataInit } });
       } else {
-        setVideos(dataSearch);
+        const response = youtube
+          .search(e.target.value)
+          .then((response) => {
+            dispatch({ type: 'grid', payload: { data: response.data } });
```

Comment:
Remember that each dispatch action (`grid` this case) have their own payload, so we could pick meaningful names for saving that data, not every action need to use the same payload key.  Just think about it.